src/transformers/modeling_tf_utils.py

```diff
@@ -1117,9 +1118,29 @@ def dummy_inputs(self) -> Dict[str, tf.Tensor]:
         Returns:
             `Dict[str, tf.Tensor]`: The dummy inputs.
         """
-        return {
-            "input_ids": tf.constant(DUMMY_INPUTS, dtype=tf.int32),
-        }
+        dummy_inputs = {}
+
+        serving_sig = self.get_serving_input_signature()
+        if self.main_input_name == "input_ids" and serving_sig[0]["input_ids"].shape.rank == 2:
+            dummy_inputs["input_ids"] = tf.constant(DUMMY_INPUTS, dtype=tf.int32)
+        elif self.main_input_name == "input_ids" and serving_sig[0]["input_ids"].shape.rank == 3:
+            dummy_inputs["input_ids"] = tf.constant(MULTIPLE_CHOICE_DUMMY_INPUTS, dtype=tf.int32)
+        elif self.main_input_name == "pixel_values":
+            image_shape = serving_sig[0]["pixel_values"].shape.as_list()
+            if image_shape[0] is None:
+                image_shape[0] = 3  # matches DUMMY_INPUTS
+            if None in image_shape[1:]:
+                raise NotImplementedError(
+                    f"Could not fully infer input tensor shape; dummy inputs or serving sig must be defined manually for {self.__class__.__name__}"
+                )
+            rng = np.random.default_rng(42)
+            VISION_DUMMY_INPUTS = rng.random(image_shape).astype(np.float32)
```

Comment:
a bit funny to define a global here which is used very locally :D

Reply:
Yeah, I'll change this! I wanted to make clear the analogy to the regular `DUMMY_INPUTS`, but vision dummy inputs will always depend on config attributes like `num_channels` and `image_size` and so they can't be generic in the same way.